### PR TITLE
Use constants for mount types

### PR DESCRIPTION
**Pull request by rhatdan**
Inspired by https://github.com/containers/podman/pull/19238

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: libpod/container_internal_common.go [hunk 1]
```diff
@@ -1962,7 +1962,7 @@ func (c *Container) makeBindMounts() error {
 			case m.Destination == "/run/.containerenv":
 				hasRunContainerenv = true
 				break Loop
-			case m.Destination == "/run" && m.Source != "tmpfs":
+			case m.Destination == "/run" && m.Source != define.TypeTmpfs:
```
Comment (Luap99) [inline libpod/container_internal_common.go:1965]: not related to this change this must compare `m.type`

Comment (baude) [reply libpod/container_internal_common.go:1965]: @Luap99 im going to merge this ... do you wanna make a quick pr afterwards to fix this?

**Pull request merged**